lib/k8s/stack.rb

```diff
@@ -101,7 +101,7 @@ def apply(client, prune: true)
         if !server_resource
           logger.info "Create resource #{resource.apiVersion}:#{resource.kind}/#{resource.metadata.name} in namespace #{resource.metadata.namespace} with checksum=#{resource.checksum}"
           keep_resource! client.create_resource(prepare_resource(resource))
-        elsif server_resource.metadata.annotations[@checksum_annotation] != resource.checksum
+        elsif server_resource.metadata.annotations.nil? || server_resource.metadata.annotations[@checksum_annotation] != resource.checksum
```

Comment:
```suggestion
        elsif server_resource.metadata&.annotations[@checksum_annotation] != resource.checksum
```
The "lonely operator" `&.` would make `nil` annotations case fail early with a falsey value and not need the explicit `.nil?` check